homeassistant/components/deconz/__init__.py

```diff
@@ -84,3 +73,33 @@ async def async_update_master_gateway(hass, config_entry):
     options = {**config_entry.options, CONF_MASTER_GATEWAY: master}
 
     hass.config_entries.async_update_entry(config_entry, options=options)
+
+
+async def async_migrate_entry(hass, config_entry) -> bool:
```

Comment:
This will make it impossible to roll back to an older version of Home Assistant after upgrading the config entry. Could we check for this case based on the config entry data instead during config entry setup?

Reply:
This is backwards compatible. The new unique id has been supported for a long time. I just didn't realize I could do it before.

Reply:
Martin you where right, I had missed to remove the CONF_GROUP_ID_BASE. This was only introduced to support the normalization of deCONZ bridgeid in order to make discovery work properly. This PR is removing the need for it since it's upgrading the entities unique IDs, by also removing the CONF in the config_entry data this will work even in previous releases

Reply:
Since we're adding `async_migrate_entry` in this PR, ie it didn't exist before version 2 of config entries for this integration, config entries with version 2 won't be able to migrate with the code of version 1.

We'll error here:
https://github.com/home-assistant/core/blob/f2d9e6f70c7f73489705a086482ac7f52d093f56/homeassistant/config_entries.py#L388-L395

Reply:
No but there is no version change done in the migrate_entry function

Reply:
I will change to not use migrate mechanism to avoid potential rollback issues